libs/k8s/custom/core/autoscaling.libsonnet

```diff
@@ -0,0 +1,32 @@
+local d = import 'doc-util/main.libsonnet';
+
+local withApiVersion = {
+  '#withApiVersion':: d.fn(help='API version of the referent', args=[d.arg(name='apiversion', type=d.T.string)]),
+  withApiVersion(apiversion): { apiVersion: apiversion },
+};
+
+
+local withScaleTargetRef = {
+  '#withScaleTargetRef':: d.fn(help='Set spec.ScaleTargetRef to `object`', args=[d.arg(name='object', type=d.T.object)]),
+  withScaleTargetRef(object):
+    { spec+: { scaleTargetRef+: withApiVersion {
```

Comment:
Merging `withApiVersion` here is adding `withApiVersion(apiversion)` as a visible key to the `ScaleTargetRef`, which then makes manifestation fail. What was the original purpose of having `withApiVersion` here?